cantools/database/utils.py

```diff
@@ -115,7 +115,13 @@ def decode_data(data: bytes,
         # remove signal that are outside available data bytes
         valid_bit_count = actual_length * 8
         for signal in fields:
-            if signal.start + signal.length > valid_bit_count:
+            if signal.byte_order == "little_endian":
+                sequential_startbit = signal.start
+            else:
+                # calculate startbit with inverted indices
+                sequential_startbit = 7 - signal.start % 8 + (signal.start // 8) * 8
```

Comment:
just wondering: this seems to be the same as [`sawtooth_to_network_bitnum()`](https://github.com/cantools/cantools/blob/master/cantools/database/utils.py#L258)?

Reply:
So that's what that function is for... yes, it's the same. The function call increases the performance penalty from ~4% to ~8% but improves readability. Should i change it?

Reply:
ok, I did not expect that consequence. Leave it then. (in C++, the performance penalty of such a function call would be zero as it would be inlined. I suppose that writing well performing code with python comes with quite a few specific pitfalls...)

Reply:
I agree. Maybe you can extend the comment with a reference to the function and why it was not used. (Only if you want.)

Reply:
done